### PR TITLE
tests(users): unit tests for user controller and services

### PR DESCRIPTION
**Pull request by timDeHof**
# Description

This PR provides unit tests for the controller and services in the users folder. 

## Issue link

Fixes #[86azu6kvp](https://app.clickup.com/t/86azu6kvp)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [x] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Test by running `yarn test:unit` or `yarn test:unit:docker` with `users` to the unit tests for users' controller and service files.

Screenshot 2024-08-06 at 7 57 15 PM

check if the unit tests meet all that is required from the [unit tests wiki page](https://github.com/chingu-x/chingu-dashboard-be/wiki/Tests#unit-tests)
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have updated the change log

### CODE REVIEW

File: src/users/users.controller.spec.ts [hunk 1]
```diff
@@ -1,25 +1,174 @@
...
+            expect(result).toEqual(usersArr);
+        });
+    });
+
+    describe("getProfile", () => {
+        it("getProfile service should be defined", async () => {
+            expect(controller.getProfile).toBeDefined();
+        });
+
+        it("should return unauthorized if user is not logged in", async () => {
```
Comment (cherylli) [inline src/users/users.controller.spec.ts:82]: I think we won't need this as we have tested the unauthorized response in e2e. But we can just leave it here.

**Review by cherylli — APPROVED** [review]
Looks good but I think we won't have to include status code tests like 400 or any tests we have already included in the e2e tests. It's not really an issue but just feels like it's repeated work. What do you think?

File: src/users/users.controller.spec.ts (hunk 1 above)
Comment (timDeHof) [reply src/users/users.controller.spec.ts:82]: Thank you for reviewing. I could remove the status code tests since they do seem like repeated work.

**Review by curtwl — APPROVED** [review]
lgtm

**Review by JoshuaHinman — APPROVED** [review]
All tests passed - and your code demonstrated some very helpful examples of unit tests!

File: src/users/users.controller.spec.ts (hunk 1 above)
Comment (timDeHof) [reply src/users/users.controller.spec.ts:82]: Hey @cherylli, 

is it ok to mark this conversation as resolved?

**Pull request merged**